Exercise-15/js/index.js

```diff
@@ -0,0 +1,84 @@
+import videoData from "../assets/video.json" assert { type: "json" };
+import posterData from "../assets/posters.json" assert { type: "json" };
+
+//-----videos section-----
+const videoContainer = document.getElementById("videoContainer");
+const videoTitle = document.getElementById("videoTitle");
+const videoDescription = document.getElementById("videoDescription");
+
+function addVideoDetails() {
+  //adding video
+  const video = document.createElement("video");
+  video.src = videoData.videoUrl;
+  video.setAttribute("width", "100%");
+  video.controls = true;
+  videoContainer.appendChild(video);
+
+  //adding video title
+  const title = document.createElement("span");
+  title.innerText = videoData.title;
+  videoTitle.appendChild(title);
+
+  //adding video description
+  const description = document.createElement("p");
+  description.innerText = videoData.description;
+  videoDescription.appendChild(description);
+}
+
+//-----comments section-----
+const commentMainContainer = document.getElementById("commentContainer");
+
+function addComments() {
+  for (let comment of videoData.comments) {
+    //image container
+    const image = document.createElement("img");
+    image.src = comment.image;
+    image.alt = comment.name;
+    const imageContainer = document.createElement("div");
+    imageContainer.className = "comment-image-container";
+    imageContainer.appendChild(image);
+
+    //details container
+    const name = document.createElement("span");
+    name.innerText = comment.name;
+    const nameContainer = document.createElement("div");
+    nameContainer.className = "comment-name";
+    const description = document.createElement("p");
+    description.innerText = comment.comment;
+    const descriptionContainer = document.createElement("div");
+    descriptionContainer.className = "comment-description";
+    const detailsContainer = document.createElement("div");
+    detailsContainer.className = "comment-details-container";
+    nameContainer.appendChild(name);
+    descriptionContainer.appendChild(description);
+    detailsContainer.appendChild(nameContainer);
+    detailsContainer.appendChild(descriptionContainer);
+
+    //main container
+    const commentContainer = document.createElement("div");
+    commentContainer.className = "comment flex";
+    commentContainer.appendChild(imageContainer);
+    commentContainer.appendChild(detailsContainer);
+
+    //adding comment
+    commentMainContainer.appendChild(commentContainer);
```

Comment:
could you let me know if the appending child to DOM element in for loop, is appropriate way

Reply:
I have used a document fragment and accessed the DOM only once